tests/ts_sql_eqc_util.erl

```diff
@@ -0,0 +1,199 @@
+%% -*- Mode: Erlang -*-
+%% -------------------------------------------------------------------
+%%
+%% Copyright (c) 2015 Basho Technologies, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+%% @doc A module to test riak_ts basic create bucket/put/select cycle.
+
+-module(ts_sql_eqc_util).
+
+-compile(export_all).
+
+%-ifdef(EQC).
+-include_lib("eqc/include/eqc.hrl").
+-include_lib("eunit/include/eunit.hrl").
+-include_lib("riak_ql/include/riak_ql_ddl.hrl").
+
+%% eqc generators
+
+%%
+%% Create Table generators
+%%
+
+gen_valid_create_table() ->
+    ?LET(DDL, gen_valid_ddl(), {DDL#ddl_v1.table, make_valid_create_table(DDL)}).
+
+%%
+%% These generators generate DDL records
+%%
+gen_valid_ddl() ->
+    ?SUCHTHAT(DDL, gen_ddl(), is_valid_ddl(DDL)).
+
+gen_ddl() ->
+    ?LET({TblName, Keys, Quantum, Fields},
+         {
+           gen_name("Table"),
+           gen_key(),
+           gen_quantum(),
+           list(gen_field())
+         },
+         #ddl_v1{table         = TblName,
+                 fields        = set_positions(Keys ++ Fields),
+                 partition_key = make_pk(Keys, Quantum),
+                 local_key     = make_lk(Keys)}).
+
+gen_key() ->
+    ?LET({Fam, Series, TS},
+         {gen_series_or_family_key_field(), gen_series_or_family_key_field(), gen_timeseries_key_field()},
```

Comment:
I thought @andytill's flexible keys now can accommodate shorter keys.  If so, should that not be a test case, too?